tests/integration/client/feedback/dataset/remote/test_dataset.py

```diff
@@ -366,6 +338,70 @@ async def test_add_metadata_property_sequential(
             assert remote_metadata_property.name == metadata_property.name
             assert len(remote_dataset.metadata_properties) == (idx + 1)
 
+    async def test_update_metadata_properties(self, owner: "User", feedback_dataset: FeedbackDataset) -> None:
+        api.init(api_key=owner.api_key)
+        workspace = Workspace.create(name="test-workspace")
+
+        remote_dataset = feedback_dataset.push_to_argilla(name="test_dataset", workspace=workspace)
+        assert isinstance(remote_dataset, RemoteFeedbackDataset)
+        assert len(remote_dataset.metadata_properties) == len(feedback_dataset.metadata_properties)
+
+        for metadata_property in remote_dataset.metadata_properties:
+            metadata_property.title = "new-title"
+            metadata_property.visible_for_annotators = False
+            remote_dataset.update_metadata_properties(metadata_property)
+        assert len(remote_dataset.metadata_properties) == len(feedback_dataset.metadata_properties)
+
+        for metadata_property in remote_dataset.metadata_properties:
+            assert metadata_property.title == "new-title"
+            assert metadata_property.visible_for_annotators is False
+
+    async def test_update_metadata_property_in_place(self, owner: "User", feedback_dataset: FeedbackDataset) -> None:
+        api.init(api_key=owner.api_key)
+        workspace = Workspace.create(name="test-workspace")
+
+        remote_dataset = feedback_dataset.push_to_argilla(name="test_dataset", workspace=workspace)
+        assert isinstance(remote_dataset, RemoteFeedbackDataset)
+        assert len(remote_dataset.metadata_properties) == len(feedback_dataset.metadata_properties)
+
+        for metadata_property in remote_dataset.metadata_properties:
+            metadata_property.title = "new-title"
+            metadata_property.visible_for_annotators = False
+            metadata_property._update()
+        assert len(remote_dataset.metadata_properties) == len(feedback_dataset.metadata_properties)
+
+        for metadata_property in remote_dataset.metadata_properties:
+            assert metadata_property.title == "new-title"
+            assert metadata_property.visible_for_annotators is False
```

Comment:
These integration tests should be written using public methods. We can rename them to check a bulk update or a one-by-one update.